core/templates/dev/head/pages/exploration_editor/simple_editor_tab/questions/MultipleChoiceInputCheckerFilter.js

```diff
@@ -0,0 +1,42 @@
+// Copyright 2014 The Oppia Authors. All Rights Reserved.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS-IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+/**
+ * @fileoverview Filter that checks whether the interaction data for a
+ * MultipleChoiceInput interaction is consistent with the simple editor.
+ */
+
+oppia.filter('MultipleChoiceInputChecker', [function() {
```

Comment:
I'm a bit confused. :) If StatesToQuestionsService uses this filter (which it doesn't actually seem to right now--nothing is using it), then won't it need to be updated in the future for other filters as new interactions are added? That doesn't seem any different in the scenario I suggested versus the one here. The service needs to be updated for new interactions either way.

If the goal is to de-couple interaction-specific behavior from StatesToQuestionsService, that sounds completely reasonable to me. I think a service is a better container for this functionality than a filter, though. Filters are specifically designed to be used in HTML, and since this is a filter it certainly could be used in HTML. If we don't want it to be usable in HTML, then I don't think it should be a filter.